msvc/unity.props

```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <PLATFORM_UNITY_DEFINES>PLATFORM_UNITY</PLATFORM_UNITY_DEFINES>
+    <PLATFORM_UNITY_INCLUDE_DIR>../../../il2cpp/build/libil2cpp/os/c-api</PLATFORM_UNITY_INCLUDE_DIR>
```

Comment:
I agree. It is not strictly a circular dependency, as Mono only depends on the OS layer in libil2cpp, and nothing in that code depends on Mono. For now, I think this is the best option, but in the future we'll pull the OS code out of libil2cpp into a separate library or repo and reference that.